src/github.com/couchbase/sync_gateway/db/design_doc.go

```diff
@@ -108,7 +124,7 @@ func (db *Database) QueryDesignDoc(ddocName string, viewName string, options map
 	// Query has slightly different access control than checkDDocAccess():
 	// * Admins can query any design doc including the internal ones
 	// * Regular users can query non-internal design docs
-	if db.user != nil && isInternalDDoc(ddocName) {
+	if db.user != nil && (isInternalDDoc(ddocName) || !db.GetUserViewsEnabled()) {
```

Comment:
Fair enough - I was on the fence about combining into a single condition in the first place.  I'll make the change.